File: Client (Angular)/models/visit-page-lists.ts
import {Diagnosis} from './diagnosis';
import {Treatment} from './treatment';
import {PreventiveTreatment} from './preventive-treatment';

export interface VisitPageLists {
  diagnosisList: Diagnosis[];
  treatmentsList: Treatment[];
  allPreventiveTreatmentsList: PreventiveTreatment[];
}

export class VisitPageListsEntity implements VisitPageLists{
  constructor() {
    this.allPreventiveTreatmentsList = [];
    this.diagnosisList = [];
    this.treatmentsList = [];
  }
  allPreventiveTreatmentsList: PreventiveTreatment[];
  diagnosisList: Diagnosis[];
  treatmentsList: Treatment[];
}

File: Server (ASP.NET Core)/Petadmin/Models/VisitPageLists.cs
using System.Collections.Generic;
using Petadmin.Core.Models;

namespace Petadmin.Models
{
    public class VisitPageLists
    {
        public List<Diagnosis> DiagnosisList { get; set; }
        public List<Treatment> TreatmentsList { get; set; }
        public List<PreventiveTreatment> AllPreventiveTreatmentsList { get; set; }
    }
}
